### PR TITLE
Register Screen frontend

### PR DESCRIPTION
**Pull request by FranRossi**
_Resolve_: [closes -CU-865ca0vrx](https://app.clickup.com/t/865ca0vrx)

## Description ⭐

I've added a Register screen that lets you fill a form a register as a PATIENT or as a DOCTOR. It uses the endpoint I've created on Laravel.

### Type of change 🏃

- [ ] Bug fix (non-breaking change which fixes an issue)
- ✅ New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots ) 📷

image

### CODE REVIEW

File: constants/navigation/NavigationProps.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export type RootStackParamList = {
+  Home: undefined;
+  Register: undefined;
+};
+
+export default function NavigationProps(): RootStackParamList {
```
Comment (rlezuez) [inline constants/navigation/NavigationProps.tsx]: why export a new `RootStackParamList` `object`? we should only need to export the `RootStackParamList` `type`

File: screens/Home/HomeScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { Button, View } from "react-native";
+import type { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { RootStackParamList } from "../../constants/navigation/NavigationProps";
+
+export default function HomeScreen({
+  navigation,
+}: NativeStackScreenProps<RootStackParamList>) {
```
Comment (rlezuez) [inline screens/Home/HomeScreen.tsx]: `NativeStackScreenProps<RootStackParamList, 'Home'>;`

File: screens/Register/RegisterScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+} from "react-native";
+import styles from "./RegisterStyle";
+import axiosConfig from "../../helpers/axios/axiosConfig";
+import type { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { RootStackParamList } from "../../constants/navigation/NavigationProps";
+import { Roles } from "../../constants/roles/Roles";
+
+export default function RegisterScreen({
+  navigation,
+}: NativeStackScreenProps<RootStackParamList>) {
```
Comment (rlezuez) [inline screens/Register/RegisterScreen.tsx]: `NativeStackScreenProps<RootStackParamList, 'Register'>;`

**Review by rlezuez — COMMENTED** [review]
other than my comments, lgtm. kudos for the usage of TypeScript

File: constants/navigation/NavigationProps.tsx (hunk 1 above)
Comment (FranRossi) [reply constants/navigation/NavigationProps.tsx]: Solved

File: screens/Register/RegisterScreen.tsx (hunk 3 above)
Comment (FranRossi) [reply screens/Register/RegisterScreen.tsx]: Solved, thanks for the comment

File: screens/Home/HomeScreen.tsx (hunk 2 above)
Comment (FranRossi) [reply screens/Home/HomeScreen.tsx]: Solved

**Pull request merged**